mobius/src/integer_arcs.rs

```diff
@@ -0,0 +1,171 @@
+    orthogonal_arcs::{compute_orthogonal_arc, compute_orthogonal_circle},
+};
+
+#[derive(Debug, Error)]
+pub enum IntegerArcError {
+    #[error("{0}")]
+    BadAngles(#[from] ArcAnglesError),
+    #[error("a and b must be distinct: {0}")]
+    DuplicateInt(i64),
+    #[error("n must be nonzero")]
+    ZeroPoints,
+    #[error("out of range value: {0} = {0}")]
+    ValueOutOfRange(String, i64),
+}
+
+/// compute a circle with diameter between a and b on the real line
+pub fn integer_circle(a: i64, b: i64) -> Result<Circle, IntegerArcError> {
+    if a == b {
+        return Err(IntegerArcError::DuplicateInt(a));
+    }
+
+    let a = a as f64;
+    let b = b as f64;
+
+    let midpoint = 0.5 * (a + b);
+    let radius = 0.5 * (a - b).abs();
+
+    Ok(Circle {
+        center: midpoint.into(),
+        radius,
+    })
+}
+
+pub fn integer_arc_by_direction(
+    a: i64,
+    b: i64,
+    direction: ArcDirection,
+) -> Result<CircularArc, IntegerArcError> {
+    let circle = integer_circle(a, b)?;
+
+    let angles = match direction {
+        ArcDirection::Counterclockwise if a < b => ArcAngles::new(PI, 2.0 * PI),
+        ArcDirection::Counterclockwise => ArcAngles::new(0.0, PI),
+        ArcDirection::Clockwise if a < b => ArcAngles::new(PI, 0.0),
+        ArcDirection::Clockwise => ArcAngles::new(0.0, -PI),
+    }?;
+
+    Ok(CircularArc { circle, angles })
+}
+
+#[derive(PartialEq)]
+pub enum Hemisphere {
+    North,
+    South,
+}
+
+pub fn integer_arc_by_hemisphere(
+    a: i64,
+    b: i64,
+    hemisphere: Hemisphere,
+) -> Result<CircularArc, IntegerArcError> {
+    let circle = integer_circle(a, b)?;
+
+    let angles = match hemisphere {
+        Hemisphere::North if a < b => ArcAngles::new(PI, 0.0),
+        Hemisphere::North => ArcAngles::new(0.0, PI),
+        Hemisphere::South if a < b => ArcAngles::new(PI, 2.0 * PI),
+        Hemisphere::South => ArcAngles::new(0.0, -PI),
+    }?;
+
+    Ok(CircularArc { circle, angles })
+}
+
+fn cyclotomic_angles(a: i64, b: i64, n: usize) -> Result<ArcAngles, IntegerArcError> {
+    if n == 0 {
+        return Err(IntegerArcError::ZeroPoints);
+    }
+
+    let n = n as i64;
+    if a >= n {
+        return Err(IntegerArcError::ValueOutOfRange("a".into(), a));
+    }
+
+    if b >= n {
+        return Err(IntegerArcError::ValueOutOfRange("b".into(), b));
+    }
+
+    if a == b {
+        return Err(IntegerArcError::DuplicateInt(a));
+    }
+    let a = a as f64;
+    let b = b as f64;
+
+    let step_size = TAU / (n as f64);
+    Ok(ArcAngles::new(a * step_size, b * step_size)?)
+}
+
+pub fn cyclotomic_circle(a: i64, b: i64, n: usize) -> Result<GeneralizedCircle, IntegerArcError> {
+    let angles = cyclotomic_angles(a, b, n)?;
+    Ok(compute_orthogonal_circle(Circle::unit_circle(), angles))
+}
+
+pub fn cyclotomic_arc_by_direction(
+    a: i64,
+    b: i64,
+    n: usize,
+    direction: ArcDirection,
+) -> Result<CircularArc, IntegerArcError> {
```

Comment:
Ideally this would return an `OrthogonalArc` that could be one of:

- `CircularArc` - most of the time
- `LineSegment` - for the diameter of the circle
- `DoubleRay` - The other half of the diameter line outside the circle

This means I'd need to make some tweaks to the underlying `compute_orthogonal_arc`